src/AuditReader.php

```diff
@@ -250,18 +258,15 @@ public function find($className, $id, $revision, array $options = [])
         }
 
         foreach ($classMetadata->associationMappings as $assoc) {
-            if (
-                ($assoc['type'] & ClassMetadata::TO_ONE) === 0
-                || false === $assoc['isOwningSide']
-                || !isset($assoc['joinColumnFieldNames'])
-            ) {
+            if (!self::isToOneOwningSide($assoc)) {
                 continue;
             }
 
-            foreach ($assoc['joinColumnFieldNames'] as $sourceCol) {
+            /** @var string $sourceCol */
+            foreach (self::getMappingValue($assoc, 'joinColumnFieldNames') as $sourceCol) {
```

Comment:
The `joinColumnFieldNames` is only available in [`ToOneOwningSideMapping`](https://github.com/doctrine/orm/blob/1153b9468ccfef452625064c66b4a135d0ea8a46/src/Mapping/ToOneOwningSideMapping.php#L26)